runtime/compiler/wasm/reader_test.go

```diff
@@ -0,0 +1,825 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright 2019-2020 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package wasm
+
+import (
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestWasmReader_readMagicAndVersion(t *testing.T) {
+
+	t.Parallel()
+
+	read := func(data []byte) error {
+		b := buf{data: data}
+		r := WASMReader{buf: &b}
+		return r.readMagicAndVersion()
+	}
+
```

Comment:
Missing happy cases:  "valid magic" and "valid magic and version"

Reply:
See below, https://github.com/onflow/cadence/pull/366/files#diff-0f55d4941b179be22cb3099b9e25051d6d10cf2bf71fd597d8d50d0007bea6ebR99